cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -465,4 +467,31 @@ int32 locate_utf8_utf8_int32(int64 context, const char* sub_str, int32 sub_str_l
   return 0;
 }
 
+#define CAST_VARCHAR(IN_TYPE, ARROW_TYPE)                                            \
+  FORCE_INLINE                                                                       \
+  const char* castVARCHAR_##IN_TYPE##_int64(int64 context, IN_TYPE value, int64 len, \
+                                            int32 * out_len) {                       \
+    arrow::internal::StringFormatter<arrow::ARROW_TYPE> formatter;                   \
+    char* ret = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, len));  \
+    if (ret == nullptr) {                                                            \
+      gdv_fn_context_set_error_msg(context, "Could not allocate memory");            \
+      return nullptr;                                                                \
+    }                                                                                \
+    arrow::Status status = formatter(value, [&](arrow::util::string_view v) {        \
+      int64 size = static_cast<int64>(v.size());                                     \
+      *out_len = static_cast<int32>(len < size ? len : size);                        \
+      memcpy(ret, v.data(), *out_len);                                               \
+      return arrow::Status::OK();                                                    \
+    });                                                                              \
+    if (!status.ok()) {                                                              \
+      return nullptr;                                                                \
```

Comment:
should we also set the error? so that any wrapping functions around this can handle the error correctly..